course03/miguel/project/spaceimpact.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "asteroidmanager.h"
+#include "spaceship.h"
+#include "utils.h"
+#include "types.h"
+
+#include <cinder/app/App.h>
+
+using namespace ci;
+using namespace ci::app;
+
+namespace SpaceImpactGame {
+
+class SpaceImpact
+{
+public:
+	SpaceImpact(const ivec2& windowSize);
+
+	void Setup();
+	void Draw() const;
+	Status Update();
+	void KeyDown(KeyEvent event);
+	void KeyUp(KeyEvent event);
+
+private:
+	const ivec2& mWindowSize;
```

Comment:
This is a bit dangerous; it's better to copy it if it's small. The risk / problem with such small (insignificant) objects is that their ownership doesnt really matter (who cares about the lifetime of a 2d vector?). If the "owner" of those coordinates ever change and the lifetime isn't at least as long as this object's lifetime, you will get a dangling ref.